api/envoy/extensions/http/stateful_session/header/v3/header.proto

```diff
@@ -28,15 +28,61 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 //
 // .. code-block:: none
 //
-//     session-header: "MS4yLjMuNDo4MA=="
+//     session-header: "MS4yLjMuNDo4MA==" # base64("1.2.3.4:80")
 //
 // When processing the upstream response, if ``1.2.3.4:80`` is indeed the final choice the extension
 // does nothing. If ``1.2.3.4:80`` is not the final choice, the new selected host will be set to
 // response headers (via the ``session-header`` response header).
 //
 // [#extension: envoy.http.stateful_session.header]
 message HeaderBasedSessionState {
-  // The name that will be used to obtain header value from downstream HTTP request or generate
-  // new header for downstream.
+  // The override mode config for the header based session state extension.
+  message OverrideModeConfig {
+  }
+
+  // The packages mode config for the header based session state extension.
+  message PackagesModeConfig {
```

Comment:
Brainstorming:
I wonder if there could be a better name than Packages mode, which is not so intuitive to me.
The closest thing I can think of is Matryoshka, but I wouldn't name it this way.
Things I can think of are related to Wrapper, Capsule (encapsulation), and Envelope.
WDYT?

Reply:
+1. Other options suggested before were: CASCADE or EXPAND.

Reply:
I have no strong point on the naming, I think Envelope sgtm.